torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NLCD dataset."""
+
+import os
+from typing import Any, Callable, Dict, List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NLCD(RasterDataset):
+    """National Land Cover Database (NLCD) dataset.
+
+    The `NLCD dataset
+    <https://www.usgs.gov/centers/eros/science/national-land-cover-database>`_
+    is a land cover product that covers the United States and Puerto Rico. It is
```

Comment:
The URLs you're using only covers CONUS, not Alaska/Hawaii/Puerto Rico. Interestingly, the 2019 version does not yet have downloads for the latter. Not sure if we should add logic to also download the latter when available or just document that this dataset is only for CONUS until someone actually needs the latter.

Reply:
I now added a statement that the current implementation is only for CONUS.